xrpl4j-model/src/main/java/org/xrpl/xrpl4j/model/transactions/NfTokenBurn.java

```diff
@@ -0,0 +1,64 @@
+package org.xrpl.xrpl4j.model.transactions;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import com.google.common.base.Preconditions;
+import org.immutables.value.Value;
+
+
+/**
+ * The {@link NfTokenBurn} transaction is used to remove an NfToken object from the
+ * NfTokenPage in which it is being held, effectively removing the token from
+ * the ledger ("burning" it).
+ *
+ * <p>If this operation succeeds, the corresponding NfToken is removed. If this
+ * operation empties the NfTokenPage holding the NfToken or results in the
+ * consolidation, thus removing an NfTokenPage, the owner’s reserve requirement
+ * is reduced by one.</p>
+ */
+@Value.Immutable
+@JsonSerialize(as = ImmutableNfTokenBurn.class)
+@JsonDeserialize(as = ImmutableNfTokenBurn.class)
+public interface NfTokenBurn extends Transaction {
```

Comment:
Just noticed this, but all of the `Transaction`s in this PR need to have a `flags` field. For transactions with no specific flags, just set it to 

```java
@JsonProperty("Flags")
  @Derived
  default TransactionFlags flags() {
    return new TransactionFlags.Builder().tfFullyCanonicalSig(true).build();
  }
```